src/sysl/importers/import_openapi.py

```diff
@@ -0,0 +1,520 @@
+
+
+def make_default_logger():
+    logger = logging.getLogger('import_openapi')
+    logger.setLevel(logging.WARN)
+    return logger
+
+
+def load_vocabulary(words_fn='/usr/share/dict/words'):
+    if not os.path.exists(words_fn):
+        return []
+    else:
+        return (w.strip() for w in open(words_fn))
+
+
+class OpenApiTranslator:
+    def __init__(self, logger, vocabulary_factory=None):
+        if vocabulary_factory is None:
+            vocabulary_factory = load_vocabulary
+        self._logger = logger
+        self._param_cache = {}
+        self._words = set()
+        self._vocabulary_factory = vocabulary_factory
+
+    def warn(self, msg):
+        self._logger.warn(msg)
+
+    def error(self, msg):
+        self._logger.error(msg)
+
+    def words(self):
+        """Lazy-load WORDS."""
+        if not self._words:
+            self._words.update(self._vocabulary_factory())
+            if not self._words:
+                self.warn("could not load any vocabulary, janky environment-specific heuristics for renaming path template names may fail")
+        return self._words
+
+    def javaParam(self, param):
+        # TODO(anz-rfc) this seems janky and fragile.
+        param = param[1:-1]
+        ident = self._param_cache.get(param)
+
+        if ident is None:
+            # foo-bar to fooBar
+            ident = re.sub(r'-(\w?)', lambda m: m.group(1).upper(), param)
+            # {fooid} -> fooId (only if foo is in WORDS but fooid isn't)
+            m = re.match(r'([a-z]+)id$', ident)
+            if m:
+                word = m.group(1)
+                if word in self.words() and word + 'id' not in self.words():
+                    ident = word + 'Id'
+            self._param_cache[param] = ident
+
+        return ident
+
+    def translate_path_template_params(self, path, params):
+        parts = re.split(r'({[^/]*?})', path)
+        if len(parts) > 1 and len(params) == 0:
+            self.warn("not enough path params path: %s" % (path))
+        pathParams = {}
+        for param in params:
+            assert param['in'] == 'path'
+            pathParams['{' + param['name'] + '}'] = param
+        for (i, p) in enumerate(parts):
+            if parts[i] in pathParams:
+                param = pathParams[parts[i]]
+                parts[i] = "{" + self.javaParam(parts[i]) + "<:" + TYPE_MAP[param['schema']['type']] + "}"
+            elif parts[i] != "" and parts[i][0] == "{":
+                self.warn("could not find type for path param: %s in params%s" % (parts[i], params))
+                parts[i] = "{" + self.javaParam(parts[i]) + "<:string}"
+        return "".join(parts)
+
+    def translate(self, oaSpec, appname, package, w):
+        hasInfo = False
+
+        hasInfo = 'info' in oaSpec
+        title = oaSpec['info'].pop('title', '') if hasInfo else ''
+
+        w(u'{}{} [package={}]:',
+            appname, title and ' ' + json.dumps(title), json.dumps(package))
+
+        with w.indent():
```

Comment:
```python
with w.indent(u'{}{} [package={}]:', appname, title and ' ' + json.dumps(title), json.dumps(package)):
```
Same everywhere else `Writer.indent()` is used.

Reply:
Changed everywhere.